### PR TITLE
Add documentation for the Go FFI design

### PR DESCRIPTION
**Pull request by aaron-congo**

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
Which tool are you using for diagrams? Is it possible to export them in vector format?
After GH compression text become almost unreadable unfortunately.

**Comment by aaron-congo** [comment]
@Yury-Fridlyand Interesting, on my computer the text is quite readable but maybe on your computer its smaller. I'm using draw.io, the diagrams I've attached are in svg format which I think is already in vector format?

File: docs/img/FFI-request-success-struct-diagram.svg
Comment (jonathanl-bq) [inline docs/img/FFI-request-success-struct-diagram.svg:1]: We need to make sure we handle non-string keys for Maps. Make sure to check against the RESP3 specification for the design of all RESP3 equivalent C types we define.

**Comment by aaron-congo** [comment]
I scaled up the diagrams and re-uploaded them. If you are looking at the diagrams in the "files changed" tab of the PR they still are a bit small but if you click "files changed" -> go to design-go-api.md and click on the three dots -> click "view file", then you can view the images embedded in the markdown file and they are easier to read

**Comment by Yury-Fridlyand** [comment]
image

**Review by jonathanl-bq — APPROVED** [review]

File: docs/img/FFI-conn-sequence.svg
Comment (Yury-Fridlyand) [inline docs/img/FFI-conn-sequence.svg:1]: Maybe add here glide-core

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**